### PR TITLE
Mosh: Update to v1.4.0

### LINKED ISSUE #4635: unable to install mosh
### Setup
_Package Name:_ mosh
_Package Version:_ 1.3.2-1

_NAS Model:_ DS1821+
_NAS Architecture:_ AMD Ryzen V1500B
_DSM version:_ DSM 6.2.4-25556

### Expected behavior
Mosh is listed as a package at https://synocommunity.com/package/mosh and should appear in the package center.

### Actual behavior
Mosh is missing from the package center.

### Steps to reproduce
_1._ Install SynoCommunity packages per instructions at https://synocommunity.com/#easy-install
_2._ Mosh doesn't appear in community tab of package center, although other SynoCommunity packages appear and are installable.
_3._ Logs are N/A, as the package is never selectable for installation.

### LINKED ISSUE #5614: [Package Request] mosh-1.4.0
### Software Name

Mosh (x64 update)

### Brief Description

Mosh already exists but does not show up for the 1522+ given its arch is R1000 (x64). Please update the mosh package to support x64.

Please see discussion in [Discord](https://discord.com/channels/732558169863225384/732559466721181738/1077255261535871046) for further information.

### Website

https://mosh.org/

### Documentation

https://github.com/mobile-shell/mosh

### Build/Installation documentation

https://github.com/mobile-shell/mosh

### Source code

https://github.com/mobile-shell/mosh

### OpenSource License

GPLv3

### Other License

_No response_

### PR DESCRIPTION
**Pull request by mreid-tt**
## Description

This PR includes the following:

1. Update Mosh to version 1.4.0

Fixes #4635, #5614

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [x] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [x] Package update
- [x] This change requires a documentation update (e.g. Wiki)

### CODE REVIEW

File: spk/mosh/Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ DESCRIPTION = mosh, the mobile shell is a remote terminal application that allow
 
 DISPLAY_NAME = mosh
 STARTABLE = no
-CHANGELOG = "1. Add libutempter to show detached sessions.<br/>2. Update openssl to v1.1.1l.<br/>3. Add Perl dependency for advanced customization and scripting."
+CHANGELOG = "1. Update Mosh to version 1.4.0."
```
Comment (hgy59) [inline spk/mosh/Makefile:15]: probably we should add the info, that openssl is updated to 1.1.1t since the last release.

Comment (mreid-tt) [reply spk/mosh/Makefile]: Done.

**Comment by mreid-tt** [comment]
Test results:

* Clean install log: https://pastebin.com/C5BZUuqR
* Upgrade install log: https://pastebin.com/JMQVs8PF
* Connected to mosh server ✅

**Review by hgy59 — APPROVED** [review]
LGTM
thank you @mreid-tt

**Pull request merged**